ui/src/views/FunctionList.tsx

```diff
@@ -0,0 +1,60 @@
+import { useGetFunctionsQuery, type Function } from "../store/generated";
+import FuncCard from "../components/Function/FuncCard";
+import { BlankSlate } from "../components/Blank";
+import { useAppDispatch } from "../store/hooks";
+import { showDocs } from "../store/global";
+import noResultsImg from "../../assets/images/no-results.png";
+
+export const FunctionList = () => {
+  const dispatch = useAppDispatch();
+
+  const { data, isFetching } = useGetFunctionsQuery();
+  const functions = data?.functions || [];
+
+  return (
+    <div className="px-5 py-4 h-full flex flex-col overflow-y-scroll">
+      <header className="mb-8">
+        <h1 className="text-lg mb-2 text-slate-50">Functions</h1>
+        <p>This is a list of all detected functions</p>
+      </header>
+
+      <div className="flex-1">
+        {isFetching ? (
+          <>Loading...</>
+        ) : functions?.length ? (
+          <div className="flex flex-col items-center basis-[36px] gap-4 max-w-xl">
+            {functions.map((f, idx) => {
+              const triggers = f?.triggers
+                ?.map((t) => t.event || t.cron)
+                .join(", ");
+              const cleanUrl = new URL(f.url || "");
```

Comment:
In the unlikely event that `f.url` is falsey, `new URL("")` will throw. Probably worth extracting this into a `useMemo()` call the conditionally setting the `contextualBar`.

Reply:
I changed the API so that URL cannot be empty from the server response. I think that should cover it in this case, but if you think not, I can try/catch this one.